### PR TITLE
[flutter_tools] Remove context from Artifacts class

### PR DESCRIPTION
**Pull request by jonahwilliams**
Also updates cache as necessary.

Removes requirement for artifacts_test.dart to use temp directory.

### CODE REVIEW

File: packages/flutter_tools/lib/src/artifacts.dart [hunk 1]
```diff
@@ -400,131 +410,174 @@ class CachedArtifacts extends Artifacts {
...
+}
+
 /// Manages the artifacts of a locally built engine.
 class LocalEngineArtifacts extends Artifacts {
-  LocalEngineArtifacts(this._engineSrcPath, this.engineOutPath, this._hostEngineOutPath);
+  LocalEngineArtifacts(
+    this._engineSrcPath,
+    this.engineOutPath,
+    this._hostEngineOutPath, {
+      @required FileSystem fileSystem,
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/artifacts.dart]: nit: indentation

File: packages/flutter_tools/test/general.shard/artifacts_test.dart [hunk 2]
```diff
@@ -3,138 +3,148 @@
...
+    });
 
-      testUsingContext('Looks up dart.exe on windows platforms', () async {
-        expect(artifacts.getArtifactPath(Artifact.engineDartBinary), contains('.exe'));
-      }, overrides: <Type, Generator>{
-        FileSystem: () => memoryFileSystem,
-        ProcessManager: () => FakeProcessManager.any(),
-        Platform: () => FakePlatform(operatingSystem: 'windows'),
-      });
+    testUsingContext('Looks up dart.exe on windows platforms', () async {
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/artifacts_test.dart]: testWithoutContext ?

File: packages/flutter_tools/lib/src/artifacts.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/artifacts.dart]: Fixed

File: packages/flutter_tools/test/general.shard/artifacts_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/artifacts_test.dart]: Fixed

**Review by zanderso — APPROVED** [review]

File: packages/flutter_tools/test/general.shard/artifacts_test.dart [hunk 3]
```diff
@@ -3,138 +3,148 @@
...
 
-      testUsingContext('Looks up dart on linux platforms', () async {
-        expect(artifacts.getArtifactPath(Artifact.engineDartBinary), isNot(contains('.exe')));
-      }, overrides: <Type, Generator>{
-        FileSystem: () => memoryFileSystem,
-        ProcessManager: () => FakeProcessManager.any(),
-        Platform: () => FakePlatform(operatingSystem: 'linux'),
-      });
+    testWithoutContext('Looks up dart on linux platforms', () async {
+      expect(artifacts.getArtifactPath(Artifact.engineDartBinary), isNot(contains('.exe')));
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/artifacts_test.dart]: doesn't this need a new `artifacts` with `platform: FakePlatform(operatingSystem: 'linux'),`?

**Review by christopherfujino — COMMENTED** [review]
one question, otherwise LGTM

File: packages/flutter_tools/test/general.shard/artifacts_test.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/artifacts_test.dart]: The platform already setup in the group is linux, so I reused that one

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/artifacts_test.dart]: ahh, missed that. lgtm.

**Review by christopherfujino — APPROVED** [review]
lgtm

**Pull request merged**